python/CMakeLists.txt

```diff
@@ -21,50 +20,59 @@ if (${Python_VERSION} VERSION_LESS 3.0.0)
   return()
 endif()
 
+add_subdirectory(src)
+
 message(STATUS "Found Python version ${Python_VERSION}. Building CUDA Quantum Python API with ${Python_INCLUDE_DIRS}")
 
 if (CMAKE_BUILD_TYPE STREQUAL "DEBUG")
   set(CMAKE_BUILD_TYPE "Debug")
 endif()
+
+# set(pybind11_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../tpls/pybind11/tools/)
+find_package(pybind11 REQUIRED PATHS ${CMAKE_CURRENT_SOURCE_DIR}/../tpls/pybind11/)
```

Comment:
I've tried this both with the setup.py at the outermost `cuda-quantum/` directory, as well as from within the `cuda-quantum/python/` directory here. Running from the outermost adds every file in the repository to the pip build, and I haven't found success manually excluding files. Instead, I've left setup.py in here, which copies over the files that I'd expect it to. 


The issue here, however, comes from linking in pybind11 to make this `pybind11_add_module` call when building the wheel. It returns the error
```
-- Found Python version 3.10.6. Building CUDA Quantum Python API with 
CMake Error at CMakeLists.txt:32 (find_package):
  Could not find a package configuration file provided by "pybind11" with any
  of the following names:

    pybind11Config.cmake
    pybind11-config.cmake

  Add the installation prefix of "pybind11" to CMAKE_PREFIX_PATH or set
  "pybind11_DIR" to a directory containing one of the above files.  If
  "pybind11" provides a separate development package or SDK, be sure it has
  been installed.
```

I've tried finding the package here as you see, and have also tried `set(pybind11_DIR <directory>)`, which also don't fix the issue.